#!/usr/bin/env python
PACKAGE = "thesis_car_drive"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("pid_kp", double_t, 0, "PID Proportional Gain", 0.1, 0.0, 2.0)
gen.add("pid_ki", double_t, 0, "PID Integral Gain", 0.01, 0.0, 1.0)
gen.add("pid_kd", double_t, 0, "PID Derivative Gain", 0.005, 0.0, 0.5)

gen.add("throttle_gain", double_t, 0, "Throttle Gain", 0.4, 0.1, 1.0)
gen.add("throttle_slope", double_t, 0, "Throttle-Velocity Slope", 18.2, 10.0, 30.0)
gen.add("throttle_offset", double_t, 0, "Throttle-Velocity Offset", 2.17, 0.0, 5.0)
gen.add("min_velocity", double_t, 0, "Minimum Velocity Threshold (m/s)", 0.5, 0.0, 2.0)

exit(gen.generate(PACKAGE, "thesis_car_drive", "ChassisParams"))